scripts/add-fields-scpca-meta.py

```diff
@@ -81,14 +87,28 @@
         print(f"No scpca-meta.json file for {run.scpca_run_id}")
         continue
 
-    # add mito file and ref fasta index if not already present
-    if all(key in results_meta for key in ('mito_file', 'ref_fasta_index')):
-        print(f"All keys are present, no updates to scpca-meta.json for {run.scpca_run_id}")
+    # create a list of new fields to check for
+    new_fields = {
+        'mito_file': "s3://scpca-references/homo_sapiens/ensembl-104/annotation/Homo_sapiens.GRCh38.104.mitogenes.txt",
+        'ref_fasta_index': "homo_sapiens/ensembl-104/fasta/Homo_sapiens.GRCh38.dna.primary_assembly.fa.fai",
+        'assay_ontology_term_id': run.assay_ontology_term_id
+    }
+
+    # check if any of the new fields are already present
+    # if they are all present, make sure that submitter_cell_types_file is up to date
+    if all(key in results_meta for key in new_fields) and results_meta['submitter_cell_types_file'] == run.submitter_cell_types_file:
+        print(f"All fields are present, no updates to scpca-meta.json for {run.scpca_run_id}")
     else:
-        results_meta['mito_file'] = "s3://scpca-references/homo_sapiens/ensembl-104/annotation/Homo_sapiens.GRCh38.104.mitogenes.txt"
-        results_meta['ref_fasta_index'] = "homo_sapiens/ensembl-104/fasta/Homo_sapiens.GRCh38.dna.primary_assembly.fa.fai"
+        # update missing fields
+        for key, default in new_fields.items():
+            results_meta.setdefault(key, default)
+
+        # update submitter cell types file if it's missing or if the value doesn't match the metadata file
+        if 'submitter_cell_types_file' not in results_meta.keys() or results_meta['submitter_cell_types_file'] != run.submitter_cell_types_file:
```

Comment:
Two "pythonic" options here:

The first is a smaller change that takes advantage of the fact that `in` always checks a dictionary's keys:
```
        if 'submitter_cell_types_file' not in results_meta or results_meta['submitter_cell_types_file'] != run.submitter_cell_types_file:
```

The second is to just use the fact that `get` returns `None` if there is no corresponding key:
```suggestion
        if results_meta.get('submitter_cell_types_file') != run.submitter_cell_types_file:
```